packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -84,115 +89,197 @@ const REFINEMENT_DISALLOW_LIST = ['c_isNew']
+    const {mutate: createCustomerProductListItem} = useShopperCustomersMutation({
+        action: 'createCustomerProductListItem'
+    })
+    const {mutate: deleteCustomerProductListItem} = useShopperCustomersMutation({
+        action: 'deleteCustomerProductListItem'
+    })
+
+    /**************** Query Actions ****************/
+    const {isLoading, isFetching, data: productSearchResult} = useProductSearch(
+        {
+            ...searchParams,
+            refine: searchParams._refine
+        },
+        {
+            keepPreviousData: true
+        }
+    )
+
+    const {error, data: category} = useCategory(
+        {
+            id: params.categoryId
+        },
+        {
+            enabled: !isSearch
+            // TODO: Why isn't this working?
+            // onError: (error) => {
+            //     const errorStatus = error.response?.status
+            //     switch (errorStatus) {
+            //         case 404:
+            //             throw new HTTPNotFound('Category Not Found.')
+            //         default:
+            //             throw new HTTPError('Unknown Error Occured.')
+            //     }
+            // }
+        }
+    )
+
+    // Apply disallow list to refinements.
+    if (productSearchResult?.refinements) {
+        productSearchResult.refinements = productSearchResult.refinements.filter(
+            ({attributeId}) => !REFINEMENT_DISALLOW_LIST.includes(attributeId)
+        )
+    }
+
+    /**************** Error Handling ****************/
+    const errorStatus = error?.response?.status
+    switch (errorStatus) {
+        case undefined:
+            // No Error.
+            break
+        case 404:
+            throw new HTTPNotFound('Category Not Found.')
+        default:
+            throw new HTTPError('Unknown Error Occured.')
+    }
+
+    /**************** Response Handling ****************/
+    if (res) {
+        res.set('Cache-Control', `max-age=${MAX_CACHE_AGE}`)
     }
 
-    const basePath = `${location.pathname}${location.search}`
     // Reset scroll position when `isLoaded` becomes `true`.
     useEffect(() => {
-        isLoading && window.scrollTo(0, 0)
-        setFiltersLoading(isLoading)
-    }, [isLoading])
+        isFetching && window.scrollTo(0, 0)
+        setFiltersLoading(isFetching)
+    }, [isFetching])
+
+    /**************** Render Variables ****************/
+    const basePath = `${location.pathname}${location.search}`
+    const showNoResults = !isLoading && productSearchResult && !productSearchResult?.hits
+    const {total, sortingOptions} = productSearchResult || {}
+    let selectedSortingOptionLabel = productSearchResult?.sortingOptions?.find(
+        (option) => option.id === productSearchResult?.selectedSortingOption
+    )
+
+    // API does not always return a selected sorting order
+    if (!selectedSortingOptionLabel) {
+        selectedSortingOptionLabel = productSearchResult?.sortingOptions?.[0]
+    }
```

Comment:
```suggestion
    const selectedSortingOptionLabel = sortingOptions?.find(
        (option) => option.id === productSearchResult?.selectedSortingOption
    ) ?? sortingOptions?.[0]
```
1. You've already destructured `sortingOptions` into a variable
2. Use `??` for the fallback value